### PR TITLE
refactor!: update MasterDetailLayout API to match web component changes

### PR DESCRIPTION
**Pull request by web-padawan**
## Summary

Fixes https://github.com/vaadin/web-components/issues/11345
Fixes https://github.com/vaadin/web-components/issues/11346
Fixes https://github.com/vaadin/web-components/issues/11347

- Remove API no longer present in the web component: `setForceOverlay`/`isForceOverlay`, `setOverlayMode`/`getOverlayMode`/`OverlayMode` enum, `masterMinSize`, and `detailMinSize` properties
- Rename `Containment` enum to `OverlayContainment` and update getter/setter (`overlayContainment` element property)
- Add new `overlaySize` property (size of detail area when shown as overlay)
- Add new `Expand` enum and `expand` property (controls which area(s) fill available space)
- Update Javadoc to align with web component JSDoc
- Update unit tests and integration test demo (`CategoryView`) to use new API

## Test plan
- [x] Unit tests pass (`mvn test -pl vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow`)
- [ ] Integration tests pass with updated web component

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/test/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayoutTest.java [hunk 1]
```diff
@@ -246,20 +241,20 @@ public void setDetailSizeWithUnit_getDetailSize() {
...
-        String minSize = "250px";
-        layout.setDetailMinSize(minSize);
-        Assert.assertEquals(minSize, layout.getDetailMinSize());
-        Assert.assertEquals(minSize,
-                layout.getElement().getProperty("detailMinSize"));
+    public void setOverlaySize_getOverlaySize() {
+        String size = "500px";
+        layout.setOverlaySize(size);
+        Assert.assertEquals(size, layout.getOverlaySize());
+        Assert.assertEquals(size,
```
Comment (vursen) [inline vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/test/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayoutTest.java:260]: Let's also test the `null` case.

Comment (web-padawan) [reply vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/test/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayoutTest.java:260]: Done.

File: vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/main/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayout.java [hunk 2]
```diff
@@ -70,24 +69,64 @@ public enum Orientation {
...
+     *            the unit for the master size
+     * @param detailSize
+     *            the size of the detail area
+     * @param detailUnit
+     *            the unit for the detail size
+     * @param expandingArea
+     *            which area(s) expand to fill available space
+     */
+    public MasterDetailLayout(float masterSize, Unit masterUnit,
+            float detailSize, Unit detailUnit, ExpandingArea expandingArea) {
```
Comment (vursen) [inline vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/main/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayout.java:126]: Looking at other components, none of them seem to have constructors with unit parameters. I think a constructor taking `String masterSize` and `String detailSize` should be enough?

Comment (web-padawan) [reply vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/main/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayout.java:126]: @rolfsmeds explicitly requested adding this option on the thread 😅  I don't have a strong opinion.

Comment (rolfsmeds) [reply vaadin-master-detail-layout-flow-parent/vaadin-master-detail-layout-flow/src/main/java/com/vaadin/flow/component/masterdetaillayout/MasterDetailLayout.java:126]: Yeah, I dunno, maybe it's overkill?

**Comment by web-padawan** [comment]
Let's wait until WC 25.2.0-alpha1 to be released before merging this PR.
